packages/catalog-process/test/CatalogService.test.ts

```diff
@@ -0,0 +1,95 @@
+import { describe, expect, it, vi } from "vitest";
+import { catalogService } from "../src/services/CatalogService.js";
+import {
+  eServiceCannotBeUpdated,
+  eServiceNotFound,
+  operationForbidden,
+} from "../src/model/domain/errors.js";
+import { EService } from "../src/model/domain/models.js";
+import { readModelGateway } from "../src/services/ReadModelGateway.js";
+import { eventRepository } from "../src/repositories/events.js";
+
+const mockEservice: EService = {
+  id: "5b041dd3-6b06-4467-9c3d-1ef11acedb88",
+  name: "name",
+  description: "description",
+  technology: "REST",
+  descriptors: [],
+  producerId: "producerId",
+  version: 1,
+};
+
+describe("CatalogService", () => {
+  describe("updateEService", () => {
+    it("updates the eservice", async () => {
+      readModelGateway.getEServiceById = vi
+        .fn()
+        .mockReturnValueOnce(mockEservice);
+
+      eventRepository.createEvent = vi
+        .fn()
+        .mockReturnValueOnce(mockEservice.id);
+
+      await expect(
+        catalogService.updateEService(mockEservice.id, mockEservice, {
+          organizationId: mockEservice.producerId,
+        })
+      ).resolves.toBeUndefined();
+      expect(eventRepository.createEvent).toHaveBeenCalledOnce();
+      expect(vi.mocked(eventRepository.createEvent).mock.lastCall)
+        .toMatchInlineSnapshot(`
+          [
+            {
+              "data": {
+                "description": "description",
+                "descriptors": [],
+                "id": "5b041dd3-6b06-4467-9c3d-1ef11acedb88",
+                "name": "name",
+                "producerId": "producerId",
+                "technology": "REST",
+                "version": 1,
+              },
+              "streamId": "5b041dd3-6b06-4467-9c3d-1ef11acedb88",
+              "type": "EServiceUpdated",
+              "version": 1,
+            },
+          ]
+        `);
+    });
+
+    it("returns an error if the eservice contains valid descriptors", async () => {
+      readModelGateway.getEServiceById = vi.fn().mockReturnValueOnce({
+        ...mockEservice,
+        descriptors: [{ state: "ARCHIVED" }],
+      });
+
+      await expect(
+        catalogService.updateEService(mockEservice.id, mockEservice, {
+          organizationId: mockEservice.producerId,
+        })
+      ).rejects.toThrowError(eServiceCannotBeUpdated(mockEservice.id));
+    });
+
+    it("returns an error if the authenticated organization is not the producer", async () => {
+      readModelGateway.getEServiceById = vi
+        .fn()
+        .mockReturnValueOnce({ mockEservice, producerId: "some-org-id" });
```

Comment:
shouldn't this be `{ ...mockEService, producerId: "..."}` with `...`?

more generally: is there type checking on these mock return values?

(I suppose, if yes, that it's more convenient on the one hand because you don't need to return a fully coherent value, but it introduces some risks in maintaining meaningful tests)